lib/features/map_view/widgets/bottom_scroll_sheet/data_list.dart

```diff
@@ -16,19 +17,21 @@ class DataSliverList<T extends GoogleNavigable> extends ConsumerWidget {
   Widget build(BuildContext context, WidgetRef ref) {
     final itemsState = ref.watch(context.mapDataController<T>());
     return switch (itemsState) {
-      AsyncLoading() => const DataListLoading(),
+      // misz-masz here fixed flickering
+      // we probably should consider similar refactoring to all switches, but won't be visible in our other use-cases
+      AsyncValue(:final IList<T?> value) =>
+        _DataSliverList<T>(value.whereNonNull.toIList()),
```

Comment:
Does the introduction of immutable list reduced flickering?

Reply:
It was one of many things I've tried cause `IList` implements list comparability, meaning `ILists` are equal if have equal elements in same order. Hovewer I forgot the riverpod doesn't check equality `==` operator anyway, but uses `identical` comparator. More you can read here in one of the questions https://riverpod.dev/docs/essentials/faq

It allowed me to reduce flickering frequency with manual check, but didn't eliminate it fully:
```dart
if (state != newstate) {
state = newstate; // or ref.invalidate or smth, basically reduces flickering to only situation when the number of places changed
} 
```



So its not the solution but I left the `IList` anyway cause it's recommended to have immutable and combarable state objects